24.0.0 fails to build with ffmpeg >= 7.0

24.0.0 fails to build with the just released ffmpeg 7.0:

```
[...]
[ 52%] Building CXX object code/CMakeFiles/code.dir/cutscene/ffmpeg/VideoDecoder.cpp.o
cd /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code && /usr/bin/x86_64-pc-linux-gnu-c++ -DANTLR4CPP_STATIC -DCMAKE_PCH -DNDEBUG -DROCKET_STATIC_LIB -DSCP_UNIX -DUSE_OPENAL -DVK_NO_PROTOTYPES -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1 -DWITH_FFMPEG -DWITH_OPENGL -DWITH_VULKAN -D_REENTRANT -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS=1 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/freespace2 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/generated_source -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/generated_source/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/OpenXR::Headers -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code/embedded_files -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/discord/src/../include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/libRocket/Include -I/usr/x86_64-pc-linux-gnu/include/freetype2 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/parsers -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/antlr4-cpp-runtime/runtime/src -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/accidental-noise -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/hidapi/hidapi -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/openxr/include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/lib/openxr/include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/mongoose -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/opengl/gl/glad/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/opengl/gl/glad_glx/include -isystem /usr/x86_64-pc-linux-gnu/include/AL -isystem /usr/x86_64-pc-linux-gnu/include/lua5.1 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/lz4 -isystem /usr/x86_64-pc-linux-gnu/include/SDL2 -isystem /usr/x86_64-pc-linux-gnu/include/ffmpeg-7 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/utfcpp -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/md5 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/libpcp/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/mdns -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/variant -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/any -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/optional/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/imgui -march=native -O2 -pipe   -W -Wall -Wextra -Wno-maybe-uninitialized -fsigned-char -Wno-unknown-pragmas -fdata-sections -ffunction-sections  -Wformat-security -Wreturn-type -Wno-deprecated -Wno-char-subscripts -Wno-stringop-truncation -Wno-stringop-overflow -O2 -Wno-unused-variable -Wno-unused-but-set-variable -Wno-array-bounds -Wno-empty-body -Wno-clobbered -Wno-unused-parameter -std=gnu++11 -fdiagnostics-color=always -Winvalid-pch -include /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code/CMakeFiles/code.dir/cmake_pch.hxx -MD -MT code/CMakeFiles/code.dir/cutscene/ffmpeg/VideoDecoder.cpp.o -MF CMakeFiles/code.dir/cutscene/ffmpeg/VideoDecoder.cpp.o.d -o CMakeFiles/code.dir/cutscene/ffmpeg/VideoDecoder.cpp.o -c /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/cutscene/ffmpeg/VideoDecoder.cpp
/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/cutscene/ffmpeg/AudioDecoder.cpp:9:30: error: 'av_get_channel_layout_nb_channels' was not declared in this scope
    9 | const int OUT_NUM_CHANNELS = av_get_channel_layout_nb_channels(OUT_CH_LAYOUT);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[...]
```

```
[...]
[ 52%] Building CXX object code/CMakeFiles/code.dir/cutscene/ffmpeg/FFMPEGDecoder.cpp.o
cd /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code && /usr/bin/x86_64-pc-linux-gnu-c++ -DANTLR4CPP_STATIC -DCMAKE_PCH -DNDEBUG -DROCKET_STATIC_LIB -DSCP_UNIX -DUSE_OPENAL -DVK_NO_PROTOTYPES -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1 -DWITH_FFMPEG -DWITH_OPENGL -DWITH_VULKAN -D_REENTRANT -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS=1 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/freespace2 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/generated_source -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/generated_source/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/OpenXR::Headers -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code/embedded_files -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/discord/src/../include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/libRocket/Include -I/usr/x86_64-pc-linux-gnu/include/freetype2 -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/parsers -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/antlr4-cpp-runtime/runtime/src -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/accidental-noise -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/hidapi/hidapi -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/openxr/include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/lib/openxr/include -I/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/mongoose -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/opengl/gl/glad/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/opengl/gl/glad_glx/include -isystem /usr/x86_64-pc-linux-gnu/include/AL -isystem /usr/x86_64-pc-linux-gnu/include/lua5.1 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/lz4 -isystem /usr/x86_64-pc-linux-gnu/include/SDL2 -isystem /usr/x86_64-pc-linux-gnu/include/ffmpeg-7 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/utfcpp -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/md5 -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/libpcp/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/mdns -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/variant -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/any -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/optional/include -isystem /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/lib/imgui -march=native -O2 -pipe   -W -Wall -Wextra -Wno-maybe-uninitialized -fsigned-char -Wno-unknown-pragmas -fdata-sections -ffunction-sections  -Wformat-security -Wreturn-type -Wno-deprecated -Wno-char-subscripts -Wno-stringop-truncation -Wno-stringop-overflow -O2 -Wno-unused-variable -Wno-unused-but-set-variable -Wno-array-bounds -Wno-empty-body -Wno-clobbered -Wno-unused-parameter -std=gnu++11 -fdiagnostics-color=always -Winvalid-pch -include /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/build/code/CMakeFiles/code.dir/cmake_pch.hxx -MD -MT code/CMakeFiles/code.dir/cutscene/ffmpeg/FFMPEGDecoder.cpp.o -MF CMakeFiles/code.dir/cutscene/ffmpeg/FFMPEGDecoder.cpp.o.d -o CMakeFiles/code.dir/cutscene/ffmpeg/FFMPEGDecoder.cpp.o -c /var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/cutscene/ffmpeg/FFMPEGDecoder.cpp
make[2]: *** [code/CMakeFiles/code.dir/build.make:1471: code/CMakeFiles/code.dir/cutscene/ffmpeg/AudioDecoder.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/cutscene/ffmpeg/FFMPEGDecoder.cpp: In function 'cutscene::ffmpeg::CodecContextParameters {anonymous}::getCodecParameters(AVStream*)':
/var/tmp/paludis/build/games-simulation-fs2_open-24.0.0/work/fs2_open-24.0.0/code/cutscene/ffmpeg/FFMPEGDecoder.cpp:72:50: error: 'AVCodecParameters' {aka 'struct AVCodecParameters'} has no member named 'channel_layout'; did you mean 'ch_layout'?
   72 |         paras.channel_layout = stream->codecpar->channel_layout;
      |                                                  ^~~~~~~~~~~~~~
      |                                                  ch_layout
make[2]: *** [code/CMakeFiles/code.dir/build.make:1487: code/CMakeFiles/code.dir/cutscene/ffmpeg/FFMPEGDecoder.cpp.o] Error 1
[...]
```
`channel_layout` is probably: https://github.com/FFmpeg/FFmpeg/commit/65ddc74988245a01421a63c5cffa4d900c47117c

--- a/code/cutscene/ffmpeg/AudioDecoder.cpp
+++ b/code/cutscene/ffmpeg/AudioDecoder.cpp
@@ -6,7 +6,7 @@ namespace {
 const int OUT_CH_LAYOUT = AV_CH_LAYOUT_STEREO;
 const int OUT_SAMPLE_RATE = 48000;
 const AVSampleFormat OUT_SAMPLE_FORMAT = AV_SAMPLE_FMT_S16;
-const int OUT_NUM_CHANNELS = av_get_channel_layout_nb_channels(OUT_CH_LAYOUT);
+const int OUT_NUM_CHANNELS = av_popcount64(OUT_CH_LAYOUT);
 
 const int DEFAULT_SRC_NUM_SAMPLES = 1024;
 

--- a/code/cutscene/ffmpeg/FFMPEGDecoder.cpp
+++ b/code/cutscene/ffmpeg/FFMPEGDecoder.cpp
@@ -69,7 +69,11 @@ CodecContextParameters getCodecParameters(AVStream* stream) {
 	paras.height       = stream->codecpar->height;
 	paras.pixel_format = (AVPixelFormat)stream->codecpar->format;
 
-	paras.channel_layout = stream->codecpar->channel_layout;
+	#if LIBAVCODEC_VERSION_INT > AV_VERSION_INT(59, 36, 255)
+		paras.channel_layout = stream->codecpar->ch_layout.u.mask;
+	#else
+		paras.channel_layout = stream->codecpar->channel_layout;
+	#endif
 	paras.sample_rate    = stream->codecpar->sample_rate;
 	paras.audio_format   = (AVSampleFormat)stream->codecpar->format;
 

